hepa-uv/firmware/hepauv_hardware.c

```diff
@@ -273,25 +312,35 @@ void set_button_led_pwm(PUSH_BUTTON_TYPE button, uint32_t red, uint32_t green, u
     button_led_hw_update_pwm(white, WHITE_LED, button);
 }
 
-void button_led_hw_initialize_leds() {
+void set_hepa_fan_pwm(uint32_t duty_cycle) {
+    // update hepa fan speed
+    htim3.Instance->CCR1 = duty_cycle;
```

Comment:
if you have this done this way, giving this function a value over the timer's period register will result in the output always being set which I think is not what you want. It will also only represent a duty cycle if the period is exactly 100. I'd recommend doing something like `htim3.Instance->CCR1 = max(duty_cycle, 100)*100 / htim3.Instance->Period` or something, actually calculate it